### PR TITLE
Timx 114 handle timdex job end

### PR DESCRIPTION
**Pull request by hakbailey**
#### What does this PR do?
Adds handling of a TIMDEX export job end post to the webhook.

#### How can a reviewer manually see the effects of these changes?
You can do a local post to the Dev1 lambda function URL to confirm that it triggers the TIMDEX step function as expected. Make sure you have the correct Dev1 credentials set for the AWS CLI and set the following variables in your .env:
```
ALMA_CHALLENGE_SECRET=<the alma challenge secret from Dev1>
LAMBDA_FUNCTION_URL=<the Dev1 lambda function URL>
VALID_TIMDEX_EXPORT_DATE="2022-10-24"
```
Then follow the instructions to send a POST request mimicking a TIMDEX export job webhook in the README and confirm both the output and that the TIMDEX step function is started (via the AWS console).

#### Includes new or updated dependencies?
YES

#### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/TIMX-114

#### Developer
- [x] All new ENV is documented in README
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer
- [ ] The commit message is clear and follows our guidelines (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

**Comment by hakbailey** [comment]
@zotoMIT tagging you primarily to review and confirm the required env variables documented in the README match what you have on the infra side.

**Comment by zotoMIT** [comment]
> @zotoMIT tagging you primarily to review and confirm the required env variables documented in the README match what you have on the infra side.

We set the following ENV, which looks like lines up with the readme.  
```
      "WORKSPACE"                          = dev/stage/prod
      "SENTRY_DSN"                         = only in stage/prod, set via ssm
      "ALMA_CHALLENGE_SECRET"              = via ssm
      "ALMA_POD_EXPORT_JOB_NAME"           = via ssm
      "ALMA_TIMDEX_EXPORT_JOB_NAME_PREFIX" = "Publishing Platform Job TIMDEX EXPORT" currently a static value
      "PPOD_STATE_MACHINE_ARN"   = set via ssm
      "TIMDEX_STATE_MACHINE_ARN" = set via ssm
    }

File: lambdas/webhook.py [hunk 1]
```diff
@@ -124,42 +113,62 @@ def valid_signature(event: dict) -> bool:
...
+    job_name = message_body["job_instance"]["name"]
+    if job_name == os.environ["ALMA_POD_EXPORT_JOB_NAME"]:
+        logger.info("PPOD export job webhook received.")
+        job_type = "PPOD"
+    elif job_name.startswith(os.environ["ALMA_TIMDEX_EXPORT_JOB_NAME_PREFIX"]):
+        logger.info("TIMDEX export job webhook received.")
+        job_type = "TIMDEX"
+    else:
+        logger.info(
+            "POST request received and validated, no action triggered. Returning 200 "
```
Comment (ehanson8) [inline lambdas/webhook.py:127]: Is there any utility to being more descriptive in the `else` log message? Not being terribly familiar with the workflow, I'm not sure if more info would be useful in troubleshooting why the `job_name` isn't lining up with POD or TIMDEX

**Review by ehanson8 — APPROVED** [review]
One inline question but the instructions were clear and the step function executed as expected

**Pull request merged**